txircd/modules/extra/webirc.py

```diff
@@ -0,0 +1,57 @@
+from twisted.plugin import IPlugin
+from twisted.words.protocols import irc
+from txircd.config import ConfigValidationError
+from txircd.module_interface import ICommand, IModuleData, Command, ModuleData
+from txircd.utils import isValidHost
+from zope.interface import implements
+
+irc.ERR_ALREADYREGISTERED = "462"
+
+class WebIRC(ModuleData, Command):
+	implements(IPlugin, IModuleData, ICommand)
+
+	name = "WebIRC"
+
+	def userCommands(self):
+		return [ ("WEBIRC", 1, self) ]
+
+	def verifyConfig(self, config):
+		if "webirc_allowed_sources" in config:
+			if not isinstance(config["webirc_allowed_sources"], dict):
+				raise ConfigValidationError("webirc_allowed_sources", "value must be a dictionary")
+			for ip, password in config["webirc_allowed_sources"]:
+				if not isinstance(ip, basestring):
+					raise ConfigValidationError("webirc_allowed_sources", "ip value must be a string")
+				if not isinstance(password, basestring):
+					raise ConfigValidationError("webirc_allowed_sources", "password value must be a string")
+
+	def parseParams(self, user, params, prefix, tags):
+		if user.isRegistered():
+			user.sendSingleError("WebircCmd", irc.ERR_ALREADYREGISTERED, "WEBIRC may only be used during registration")
+			return None
+		if len(params) < 4:
+			user.sendSingleError("WebircCmd", irc.ERR_NEEDMOREPARAMS, "WEBIRC", "Not enough parameters")
+			return None
+		if user.ip not in self.ircd.config.get("webirc_allowed_sources", {}) and user.realHost not in self.ircd.config.get("webirc_allowed_sources", {}):
+			self.ircd.log.warn("WEBIRC was requested from IP \"{ip}\", but the IP does not match any WEBIRC configuration.", ip=user.ip)
+			return None
+		if self.ircd.config["webirc_allowed_sources"][user.ip] != params[0]:
+			self.ircd.log.warn("WEBIRC was requested from IP \"{ip}\" with password \"{password}\", but this password does not match the WEBIRC configuration for this IP.", ip=user.ip, password=params[0])
+			return None
```

Comment:
I'm not familiar with the WEBIRC spec or implementations of it, but should there be some sort of feedback on a failed password, or no?

I'm totally fine with not, but I want to make sure this is expected.